doc/cephadm/upgrade.rst

```diff
@@ -188,3 +188,43 @@ you need. For example, the following command upgrades to a development build:
   ceph orch upgrade start --image quay.io/ceph-ci/ceph:recent-git-branch-name
 
 For more information about available container images, see :ref:`containers`.
+
+
+Migrations
+==========
+
+.. _cephadm-migration-nfs:
+
+16.2.7 - NFS export migration
+-----------------------------
+
+This migration should be done fully automated, but in case something goes
+wrong while the upgrade is in progress, there is also a manual procedure
+
+in 16.2.6, the pool for storing NFS ganesha exports was arbitrary, after
+16.2.7, the pool name is always ``.nfs``. to create the Pool, run::
+
+  osd pool create pool .nfs
+  osd pool application enable .nfs nfs
+
+Previously, there was an extra parameter which was the RAODOS namespace. 
+After 16.2.7, the namespace is now always set to the cephadm service id. 
+
+Next, we have to copy over the ``grace`` object from the old location to
+the new location.
+
+To migrate export objects (all RADOS objects starting with ``export-``),
+run::
+
+  nfs export apply --cluster-id <cephadm-service-id> -i <local-path-to-export-object>
```

Comment:
```suggestion
  ceph nfs export apply --cluster-id <cephadm-service-id>
```

can we use `nfs export ls <cluster_id> --detailed` to generate a json list of all exports before hand?

Reply:
> can we use `nfs export ls <cluster_id> --detailed` to generate a json list of all exports before hand?

I don't think so. The nfs module only works the new way reliably.